### PR TITLE
Added new default functions for `configauth` file

### PR DESCRIPTION
**Pull request by AkhigbeEromo**
Description:
Added NewDefault methods for structs in configauth package

Link to tracking Issue:
Closes https://github.com/open-telemetry/opentelemetry-collector/issues/9821

Testing: Tests were added for the NewDefault functions

### CODE REVIEW

**Comment by AkhigbeEromo** [comment]
Hello @jpkrohling, can i please get a review

File: config/configauth/configauth_test.go [hunk 1]
```diff
@@ -15,6 +15,11 @@ import (
 
 var mockID = component.MustNewID("mock")
 
+func TestNewDefaultAuthentication(t *testing.T) {
+	auth := NewDefaultAuthentication()
+	assert.NotNil(t, auth)
```
Comment (TylerHelmuth) [inline config/configauth/configauth_test.go:20]: Lets require that `AuthenticatorID` not be set.

Comment (AkhigbeEromo) [reply config/configauth/configauth_test.go:20]: how do I do this?
tried setting it to an empty string but it keeps throwing an error.

Comment (TylerHelmuth) [reply config/configauth/configauth_test.go:20]: Something like `assert.Empty(t, auth)`

Comment (AkhigbeEromo) [reply config/configauth/configauth_test.go:20]: Thank you
There are no errors again
Will push the commit now

**Review by TylerHelmuth — APPROVED** [review]

**Comment by AkhigbeEromo** [comment]
I am having the same error here, as I did in the last PR I created, so I am going to submit another one @TylerHelmuth

**Pull request closed without merging**